### ISSUE TITLE
How to delete a rule?

### ISSUE THREAD

**Issue by hongfeiyucode** [phhusson/sepolicy-inject#11]
I mean **Completely remove the rules** and **Make the file smaller**.
Rather than modifying the flag.

**Comment by hongfeiyucode** [comment]
I do not understand the principle：

```av->data &= ~(1U << (p - 1));```

**Comment by pppdddxxx** [comment]
Yeah! I also want to know **how to remove some sepolicy, that make the sepolicy file smaller** . thanks

**Comment by phhusson** [comment]
How would you know which rule to delete?

**Comment by hongfeiyucode** [comment]
For example：I want to delete all the neverallow rules.
Of course, it can be other rules.
As the case may be.

**Comment by phhusson** [comment]
neverallow is NOT stored inside sepolicy
It is only a build-time parameter

**Comment by hongfeiyucode** [comment]
How about deleting a specified rule?

**Comment by phhusson** [comment]
If the aim is to get place, you don't need to delete just one rule, but a full class of rules.
There is already a command to remove a specific rule, but as you pointed out, it won't gain you any space.

Again, how would you know which class of rules do you want to delete?
For your goal, isn't disabling SELinux good enough?

**Comment by hongfeiyucode** [comment]
Yes, my purpose is this.
I would like to inject some rules at the same time to ensure that the file size will not become larger.

**Comment by pppdddxxx** [comment]
@phhusson , Hi
My perpose is to **inject some new policy in sepolicy and don't let the file become bigger.**

**Comment by hongfeiyucode** [comment]
@pppdddxxx Do you know how to do it now?

**Comment by hongfeiyucode** [comment]
@phhusson 
It is not difficult to delete rules.
The difficulty is to protect the sepolicy file structure.
After deleting the rules, the file structure is corrupted and can not be recognized properly.

**Comment by pppdddxxx** [comment]
@phhusson Can I replace some policy?
